test/Microsoft.AspNetCore.SignalR.Redis.Tests/RedisServerFixture.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.SignalR.Tests.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Tests
+{
+    public class RedisServerFixture<TStartup> : IDisposable
+        where TStartup : class
+    {
+        public ServerFixture<TStartup> FirstServer { get; private set; }
+        public ServerFixture<TStartup> SecondServer { get; private set; }
+
+        private readonly ILogger _logger;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IDisposable _logToken;
+
+        public RedisServerFixture()
+        {
+            if (Docker.Default == null)
+            {
+                return;
+            }
+
+            var testLog = AssemblyTestLog.ForAssembly(typeof(RedisServerFixture<TStartup>).Assembly);
+            _logToken = testLog.StartTestLog(null, $"{nameof(RedisServerFixture<TStartup>)}_{typeof(TStartup).Name}", out _loggerFactory, "RedisServerFixture");
+            _logger = _loggerFactory.CreateLogger<RedisServerFixture<TStartup>>();
+
+            FirstServer = new ServerFixture<TStartup>();
+            SecondServer = new ServerFixture<TStartup>("http://localhost:3123");
```

Comment:
why you specify url only for one of the servers. Also what happens if the url is taken?

Reply:
ServerFixture starts the server in the ctor - you may want to add some exception handling/logging so that it is easy to figure out what just happened.